src/menu/menu.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef, useState } from "react";
+import { PopoverTrigger } from "../popover-v2";
+import { MenuItem } from "./menu-item";
+import { MenuSection } from "./menu-section";
+import { MenuPanel } from "./menu.styles";
+import { MenuProps } from "./types";
+import { MenuLink } from "./menu-link";
+
+export const MenuTrigger = ({
+    children,
+    position = "bottom-start",
+    onPopoverAppear,
+    menuContent,
+    scrollable,
+    "data-testid": testId = "menu",
+    className,
+    ...otherProps
+}: MenuProps): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const panelRef = useRef<HTMLDivElement | null>(null);
+    const [maxHeight, setMaxHeight] = useState<number>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        if (scrollable) {
+            window.addEventListener("resize", handleViewportChange);
+            window.addEventListener("scroll", handleViewportChange);
+            return () => {
+                window.removeEventListener("resize", handleViewportChange);
+                window.removeEventListener("scroll", handleViewportChange);
+            };
+        }
+    }, [scrollable]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleAppear = () => {
+        onPopoverAppear?.();
+        scrollable &&
+            requestAnimationFrame(() => {
+                handleViewportChange();
+            });
+    };
+
+    const handleViewportChange = () => {
+        if (panelRef.current) {
+            const rect = panelRef.current.getBoundingClientRect();
+            setMaxHeight(window.innerHeight - rect.top);
+        }
+    };
```

Comment:
handles scroll behaviour of the menu (this needs to be used with any of the position bottoms, and enableFlip = false).

probably good to refactor this into a util/custom hook, so that it can also be used by dropdowns and popover.
(i see a similar issue with dropdowns - not sure if fixed already, should the dropdown is at the bottom of the screen, it extends past the viewport)

Reply:
for scroll behaviour, what about the size middleware? https://floating-ui.com/docs/size

however, there's a limitation with size in that all floating props needs to be passed down to the scrollable container, else the height computation doesn't work well and breaks the flip logic

I'm using it here with flip in `element-with-dropdown` and I'm still looking to improve the default behaviour, but it seems that exposing the floating props is unavoidable
https://github.com/LifeSG/react-design-system/blob/improve-auto-positioning/src/shared/dropdown-wrapper/element-with-dropdown.tsx#L156

Reply:
updated/refactored to use "size" into popover content (thought that popover content should also have this same behaviour) (updated storybook to see all the variations)